examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.examples.snappydata
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.log4j.{Level, Logger}
+
+import org.apache.spark.sql.{SnappyContext, SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession, SparkSession}
+
+/**
+ * This is a sample code snippet to work with domain objects and SnappyStore column tables.
+ * Run with
+ * <pre>
+ * bin/run-example snappydata.WorkingWithObjects
+ * </pre>
+ */
+
+case class Address(city: String, state: String)
+
+case class Person(name: String, address: Address)
+
+object WorkingWithObjects extends SnappySQLJob {
+
+  override def isValidJob(snc: SnappyContext, config: Config): SnappyJobValidation = SnappyJobValid()
+
+  override def runSnappyJob(snc: SnappyContext, jobConfig: Config): Any = {
+
+
+    //Import the implicits for automatic conversion between Objects to DataSets.
+    import snc.implicits._
+
+    val snSession = snc.snappySession
+    // Create a Dataset using Spark APIs
+    val people = Seq(Person("Tom", Address("Columbus", "Ohio")), Person("Ned", Address("San Diego", "California"))).toDS()
+
+
+    //Drop the table if it exists.
+    snSession.dropTable("people", ifExists = true)
+
+    // Write the created Dataset to a column table.
+    people.write
+        .format("column")
+        .options(Map("BUCKETS" -> "1", "PARTITION_BY" -> "name"))
+        .saveAsTable("people")
```

Comment:
Firstly remove this implicit table creation and buckets specification. Lets demo the normal CREATE TABLE SQL or createTable API throughout. Use insertInto().

As in the SNAP-490 branch we would like to also show direct RDD insertInto for case class objects. When it is merged please take note to add something like:

```scala
import snSession.sqlImplicits._ // use this instead of snc.implicits._
...
snSession.createTable("people", ...)
people.insertInto("people")
```

This is both more efficient and cleaner for end-users.